src/Aspire.Hosting.Azure/Provisioning/Provisioners/BicepProvisioner.cs

```diff
@@ -176,15 +176,8 @@ await notificationService.PublishUpdateAsync(resource, state =>
             Parameters = BinaryData.FromObjectAsJson(parameters),
             DebugSettingDetailLevel = "ResponseContent"
         });
-        // Only set the location and custom deployment name
-        // for top-level AzureEnvironmentResource.
-#pragma warning disable ASPIREAZURE001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
-        if (resource is AzureEnvironmentResource)
-        {
-            deploymentContent.Location = context.Location;
-            deploymentName = $"{resourceGroup.Name}-{DateTimeOffset.Now.ToUnixTimeSeconds()}";
-        }
-#pragma warning restore ASPIREAZURE001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+        // Set the location and use timestamped deployment name for all resources
+        deploymentName = $"{resource.Name}-{DateTimeOffset.Now.ToUnixTimeSeconds()}";
```

Comment:
Now that we are not deploying main.bicep, we might not need to disambiguate each resource deployment with a timestamp but I'm keeping it here while we evaluate.